### PR TITLE
IoTCentral: Add and swap to embedded SDK

### PR DESCRIPTION
**Pull request by stephybun**

### CODE REVIEW

File: internal/services/iotcentral/iotcentral_application_resource.go
Comment (tombuildsstuff) [inline internal/services/iotcentral/iotcentral_application_resource.go:77]: F1 isn't present here?

File: internal/services/iotcentral/iotcentral_application_resource.go
Comment (tombuildsstuff) [inline internal/services/iotcentral/iotcentral_application_resource.go:79]: fwiw these should be:

```suggestion
				Default: string(apps.AppSkuSTOne),
```

File: internal/services/iotcentral/iotcentral_application_resource.go
Comment (tombuildsstuff) [inline internal/services/iotcentral/iotcentral_application_resource.go:115]:
```suggestion
	resp, err := client.CheckNameAvailability(ctx, commonids.NewSubscriptionID(id.SubscriptionId), inputs)
```

File: internal/services/iotcentral/iotcentral_application_resource.go
Comment (tombuildsstuff) [inline internal/services/iotcentral/iotcentral_application_resource.go:232]:
```suggestion
	id, err := apps.ParseIotAppID(d.Id())
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
Left a few comments inline - will follow up offline 👍

File: internal/services/iotcentral/iotcentral_application_resource.go
Comment (stephybun) [reply internal/services/iotcentral/iotcentral_application_resource.go:77]: The new version of the SDK doesn't have F1 or S1 anymore. Even though the older SDK has F1 and S1 as valid enums the API will throw an error if you try and create this resource with F1 or S1.

**Comment by hmmorales** [comment]
Hello is there an expected date for this PR? We are curious because [16899](https://github.com/hashicorp/terraform-provider-azurerm/pull/16899) is blocked until this PR is merged.

**Pull request merged**
